### PR TITLE
Use "esm" for ECMAScript Imports at Buildtime

### PR DESCRIPTION
**Pull request by jestapinski**
This PR uses a package called [`esm`](https://github.com/standard-things/esm) to allow us to use more modern import syntax at buildtime. This was surprisingly painless to add!

### CODE REVIEW

File: gatsby-node.esm.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
+const path = require('path');
```
Comment (jestapinski) [inline gatsby-node.esm.js:1]: There are no code differences between here at the previous `gatsby-node.js`. I just moved all of that code into this file for `esm`.

File: gatsby-node.js [hunk 2]
```diff
@@ -1,168 +1,2 @@
...
-    onCreatePage(
-        page,
-        actions,
-        stitchClient,
-        homeFeaturedArticles,
-        learnFeaturedArticles,
-        excludedLearnPageArticles
-    );
+require = require('esm')(module);
+module.exports = require('./gatsby-node.esm.js');
```
Comment (jestapinski) [inline gatsby-node.js:2]: Here is the `esm` magic, suggested in [this Gatsby issue thread](https://github.com/gatsbyjs/gatsby/issues/7810#issuecomment-449741977)

File: gatsby-node.esm.js (hunk 1 above)
Comment (greggb) [reply gatsby-node.esm.js:1]: In a future PR we can move all the `require`s and `exports` to es syntax 🎉

**Comment by jestapinski** [comment]
Thanks! Yes I plan to address the es syntax shortly

**Pull request merged**